File: testes/gerarAnagrama.py
import random

def gerarAnagrama():
    '''Esta funcao gera um anagrama para uma palavra passada por input'''
    palavra = input('Digite uma palavra: ')
    lista = list(palavra)

    i=0
    saida = ''

    # a quantidade de vezes que o laco se repete eh um valor arbitrario
    # len(lista)*5 foi escolhido por nao ser um valor muito alto, mas tambem nao muito pequeno
    
    while (i<len(lista)*5):
        # aqui sao definidas as posicoes dos elementos a serem permutados
        primeiro = random.randrange(0,len(lista))
        segundo = random.randrange(0, len(lista))
        lista[primeiro], lista[segundo] = lista[segundo], lista[primeiro]
        i+=1

    for letra in lista:
        # construcao da string de saida
        saida += letra
    
    return print('Um dos anagramas de %s eh %s' %(palavra, saida))

gerarAnagrama()

File: testes/gerarAnagrama.c
#include                        <stdio.h>
#include                        <stdlib.h>
#include                        <time.h>
#include                        <string.h>

#define OK                      0
#define TAMANHO_MAX             50

int main()
/** gera e imprime um anagrama para uma palavra passada por input */  
{
    char palavra[TAMANHO_MAX], auxiliar, saida[TAMANHO_MAX];
    int indice, teto, primeiro, segundo;

    srand(time(NULL));

    for (;;)
    {
        // Pegando a palavra
        printf ("Digite a palavra: ");
        scanf("%[^\n]%*c", palavra);
        teto = strlen(palavra);
        if (teto > TAMANHO_MAX)
            printf ("A palavra excede o tamanho maximo de %d caracteres\n", TAMANHO_MAX);
        else
            break;
    }

    /* Copiando a palavra de entrada para a saida, 
     * para poder permutar as letras da saida sem alterar a original.
     * A variavel teto serve para delimitar o maior inteiro a ser gerado
     * no processo de permutacao, para evitar erros inesperados */
    strcpy(saida, palavra);

    /* o valor escolhido para o fim do laco eh a quantidade de permutacoes que ocorrerao
     *   este valor eh completamente arbitrario */
    for (indice = 0; indice<teto*5; indice++)
    {
        // definindo as posicoes a serem permutadas
        primeiro = (rand()%(teto));
        segundo = (rand()%(teto));

        // permutando os valores
        auxiliar = saida[primeiro];
        saida[primeiro] = saida[segundo];
        saida[segundo] = auxiliar;
    }
    printf ("Um anagrama para %s eh %s\n", palavra, saida);
    return OK;
}